activerecord/lib/active_record/associations/builder/singular_association.rb

```diff
@@ -21,11 +22,7 @@ def reload_#{name}
         end
 
         def #{name}!
-          association = association(:#{name}).force_reload_reader
-
-          raise ActiveRecord::RecordNotFound if association.nil?
-
-          association
+          association(:#{name}).owner.send("#{plural_name}").sole
```

Comment:
I don't know if this is the _best_ way to do it, but it got my simple tests to pass.